src/isa/prog.cc

```diff
@@ -282,18 +308,35 @@ void top_k_progs::clear() {
 }
 
 bool top_k_progs_sort_function(prog* x, prog* y) {
-  return (x->_perf_cost < y->_perf_cost);
+  if(logger.k2_functionality == FUNC_optimize){
+    return (x->_perf_cost < y->_perf_cost);
+  }else if(logger.k2_functionality == FUNC_repair){
+    return (x->_safety_cost < y->_safety_cost);
+  }else{
```

Comment:
Could you please format the code by adding a space between `if` and `{`, `}` and `else`, `}` and `else`?
There may be other code that needs to be formatted. Could you please kindly check it? Thanks!